### DISCUSSION TITLE
Group control elements in serenity dialog/form

### DISCUSSION THREAD

**Discussion by vmrosariojoe** [serenity-is/Serenity#5395] [Help] [answered]
Hi Team, my screen dialog output appears like this :
![20](https://user-images.githubusercontent.com/22878845/105702818-4cc18600-5f47-11eb-861d-30ac396467cc.PNG)
![21](https://user-images.githubusercontent.com/22878845/105702931-78dd0700-5f47-11eb-9b1f-f0ff6a361e0d.PNG)

I want the UI to group the elements and appear like below :
![22](https://user-images.githubusercontent.com/22878845/105704930-76c87780-5f4a-11eb-9020-82b2ed80217f.PNG)
![23](https://user-images.githubusercontent.com/22878845/105704943-7b8d2b80-5f4a-11eb-9694-1441e5713063.PNG)
How to achieve this in my form/dialog with css attributes?

**Comment by minhhungit** [comment]
Hi @vmrosariojoe  try something like this on your form.cs

```csharp
[HalfWidth]
public String PatientReferenceNo{ get; set; }

[HalfWidth]
public String PatientFullName{ get; set; }

[HalfWidth]
[FormCssClass("line-break-sm")]
public DateTimePatientDateOfBirth{ get; set; }

[HalfWidth]
[FormCssClass("line-break-sm")]
public int Age{ get; set; }

[HalfWidth]
[FormCssClass("line-break-sm")]
public String PatientPatientImage{ get; set; }
```

Put [HalfWidth] on all fields, but just put `[FormCssClass("line-break-sm")]` from 3rd field to last field


Read more here
https://serenity.is/blog/2018/05/16/multi-column-layout

**Comment by vmrosariojoe** [comment]
minhhungit,
Thanks for your reply. However i'm not able to get the desired output. Still there is space:
![25](https://user-images.githubusercontent.com/22878845/105714419-489d6480-5f57-11eb-9e1f-9a21df4fedc3.png)

My code : 

        [DisplayName("PRN")]
        [HalfWidth]
        public Int64? PatientId { get; set; }

        [HalfWidth]
        [ReadOnly(true)]
        [DisplayName("Patient Image")]
        public StringField PatientPatientImage { get; set; }

        [DisplayName("Reference No")]
        [ReadOnly(true)]
        [HalfWidth]
        [FormCssClass("line-break-sm")]
        public StringField PatientReferenceNo { get; set; }

        [DisplayName("Full Name")]
        [ReadOnly(true)]
        [HalfWidth]
        [FormCssClass("line-break-sm")]
        public StringField PatientFullName { get; set; }

        [DisplayName("Date Of Birth")]
        [ReadOnly(true)]
        [HalfWidth]
        [DisplayFormat("d")]
        [FormCssClass("line-break-sm")]
        public DateTimeField PatientDateOfBirth { get; set; }

        [DisplayName("Age")]
        [ReadOnly(true)]
        [QuarterWidth]
        [FormCssClass("line-break-sm")]
        public StringField Age { get; set; }
        
        [QuarterWidth]
        [ReadOnly(true)]
        [DisplayName("Gender")]
        [FormCssClass("line-break-sm")]
        public StringField GenderDescription { get; set; }

**Comment by minhhungit** [comment, marked as answer]
Hi @vmrosariojoe, to custom dialog layout, you can add a html template file with name same with your dialog, for example, your dialog has name `AbcDialog.ts` then create html template with name `AbcDialog.Template.html` 

![image](https://user-images.githubusercontent.com/2279508/105795556-a1084c80-5fbf-11eb-8fbc-16a1503c3cb9.png)

**CategoryDialog.Template.html**
```html
<div class="s-DialogContent">
    <div id="~_Toolbar" class="s-DialogToolbar">
    </div>
    <div class="s-Form">
        <form id="~_Form" action="">
            <div class="fieldset" style="display: block">
                <div id="~_PropertyGrid" class="col-md-6"></div>
                <div id="~_rightPanelContainer" class="col-md-6"></div>
                <div class="clear"></div>
            </div>
        </form> 
    </div>
</div>
```

The template above add a div on right, we will use it to hold Image field

Edit your dialog.ts and move your Image field to new container

![image](https://user-images.githubusercontent.com/2279508/105795733-e462bb00-5fbf-11eb-946a-d20cb5bb5485.png)


```javascript
constructor() {
      super();
      this.byId("rightPanelContainer").append(this.form.Picture.getGridField());
  }
```

Nothing change on your Form.cs, but your field should be [FullWidth] instead [HaftWidth] like before, because right now they just take 50% of dialog

!!! Keep in mind that this is hack and it maybe a bug in future, so **DO IT AT YOUR OWN RISK**

Good luck, here is demo
![image](https://user-images.githubusercontent.com/2279508/105795418-62729200-5fbf-11eb-9660-4a1df14f5ea3.png)

**Reply by vmrosariojoe** [reply]
Thanks a lot minhhungit. This is working. Much appreciated.
